plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/transition/MapboxViewportTransitionFactory.kt

```diff
@@ -0,0 +1,278 @@
+package com.mapbox.maps.plugin.viewport.experimental.transition
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.view.animation.Interpolator
+import androidx.core.view.animation.PathInterpolatorCompat
+import com.mapbox.geojson.Point
+import com.mapbox.maps.CameraOptions
+import com.mapbox.maps.EdgeInsets
+import com.mapbox.maps.plugin.animation.CameraAnimatorOptions.Companion.cameraAnimatorOptions
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.viewport.experimental.ViewportPluginImpl.Companion.VIEWPORT_CAMERA_OWNER
+import kotlin.math.abs
+
+/**
+ * Helper class that provides default implementation of [ViewportTransition] generators.
+ */
+internal class MapboxViewportTransitionFactory(
+  mapDelegateProvider: MapDelegateProvider
+) {
+  private val cameraManager = mapDelegateProvider.mapCameraManagerDelegate
+  private val cameraPlugin = mapDelegateProvider.mapPluginProviderDelegate.camera
+
+  fun transitionFromLowZoomToHighZoom(
+    cameraOptions: CameraOptions,
+    transitionMaxDurationMs: Long
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
```

Comment:
btw high-level q - we do want to use regular `Animator`s here, correct? and not our `CameraAnimator`

Reply:
ah, I see that center animator is created below so perhaps to avoid confusion we could indeed use `CameraAnimator` here and not Android SDK `Animator`

Reply:
These are ported from nav sdk, we could refactor these, will take a deeper look in next steps.